### DISCUSSION TITLE
Converting a string into cocotb handle

### DISCUSSION THREAD

**Discussion by PiyushSaini97** [cocotb/cocotb#4256] [Q&A]
Hi,

I've a really odd question.
Is it possible for me to declare a signal hierarchy as a string and then convert it back to handle so that I can drive a value on it.

For example:
`signalPath = "dut.block1_inst.block2_inst.signal"`

what do I do now so that I can write `signalPath.value = 1`

I did it via a very crooked way - 
```
signalPathStr = "dut.block1_inst.block2_inst.signal"
signalPath = signalPathStr.split(".",1)

for path in signalPath[1:]:
            signalHandle = getattr(cocotb.top,path)

signalHandle.value = 1
```

For some reason this does not force the value to what I want in TB.
Am i missing something?

--------------------------------------------
cocotb version 1.8.1
python v3.9.0

**Comment by ktbarrett** [comment]
You can `eval` the string? This should also make it work when 

```python
signalPathStr = "dut.block1_inst.block2_inst.signal"
eval(signalPathStr, locals=dict(dut=cocotb.top)).value = 1
```

You could even wrap that into a simple function:
```python
def getHandleFromPathString(signalPathStr: str) -> SimHandleBase:
    return eval(signalPathStr, locals=dict(dut=cocotb.top))

getHandleFromPathString(signalPathStr).value = 1
```

At some point in the future we might extent the handle indexing syntax to support taking full paths, but haven't taken that jump yet since it would require parsing the pathspec (slow).
